### ISSUE TITLE
About the results of two frames

### ISSUE THREAD

**Issue by heiheishuang** [SysCV/r3d3#1]
Hello, I'm very interested in your work! I have a question, have you tried training the network just using two frames? I've modify some of the settings and ended up with a result that doesn't look too good. May I ask if this result is normal?

Here are my modifications:
```
r3d3_n_warmup = 2
r3d3_optm_window = 1
r3d3_dt_intra = 1
r3d3_dt_inter = 1
```

And my results:
![Screenshot from 2023-09-11 20-43-39](https://github.com/SysCV/r3d3/assets/49268719/6cf8510c-d8bb-4886-a38b-56782d194fb4)

**Comment by AronDiSc** [comment]
Hi @heiheishuang, thank you for your interest in our work! 

First, have you applied those modifications to step "1. Generate Training Data"? If so, it might happen that the generated poses are bad. You can check the ATE of the respective scene where you took your two frames from when generating the data. Note though that we've deliberately chosen a large warmup, optimization window, etc. to get accurate poses for training and because runtime does not matter for generating training data. Also, note that those modifications do not have an impact on the training itself (apart from the generated data), but only on the inference pipeline (i.e. running evaluate.py).

Second, am I correctly understanding that you've tried to overfit the network on two timesteps with all six frames? And you evaluate on the same timesteps/frames? Because depth is implicitely learned through the photometric loss, it greatly depends which frames you've chosen (how is the texture and lighting in those frames, are there many dynamic objects, etc.). Furthermore, the photometric loss makes it nearly impossible for the learned depth to converge to ground truth. In fact, self-supervised depth learning with the photometric loss benefits from many samples, i.e. you should get better results when training on more than just two samples.

In conclusion, depending on the chosen frame, the accuracy of the poses and the number of epochs trained, your results can be reasonable.

I've also run a quick overfitting experiment on a single timestamp (DDAD scene 000000 - 20th sample). After training 8 epochs I get the following results (I train each epoch with 100 copies of the respective sample).

![image](https://github.com/SysCV/r3d3/assets/38748463/5ee4f16a-9eb0-433f-bc36-a6c9f43a838d)

To get the same results you can try to ...
1. Run step "1. Generate Training Data" for DDAD. You can terminate after the data for scene "000000" is generated
2. Make a new file (e.g. `one_sample_split.json`) with the content `{"000000": [20]}`
3. Make a copy of `configs/training/depth_completion/r3d3_completion_ddad_stage_2.yaml`, rename to `r3d3_completion_ddad_one_sample.yaml` and add the following content to the training and validation sections
```
datasets:
    train:
        repeat: [ 100 ]
        split: [ <path_to_json>/single_file_train.json ]
        ...
    validation:
        split: [  <path_to_json>/single_file_train.json ]
        ...
```
4. Run `python train.py <path_to_config>/r3d3_completion_ddad_one_sample.yaml`

I hope this helps you.

**Comment by heiheishuang** [comment]
Thank you so much for the reply!
